examples/vision/fully_convolutional_network.py

```diff
@@ -0,0 +1,631 @@
+    zip(fcn32s_dict.items(), fcn16s_dict.items(), fcn8s_dict.items())
+):
+    plt.subplot(rows, cols, pos[i])
+    plt.plot(range(len(value_32s)), value_32s)
+    plt.plot(range(len(value_16s)), value_16s)
+    plt.plot(range(len(value_8s)), value_8s)
+    plt.title(str(key_32s) + " (combined)")
+    plt.legend(["FCN-32S", "FCN-16S", "FCN-8S"])
+
+plt.show()
+
+"""
+### Visualizing predicted segmentation masks
+
+To understand the results and see them better, we pick a random image from the test
+dataset and perform inference on it to see the masks generated by each model.
+"""
+
+images, masks = next(iter(test_ds))
+random_idx = tf.random.uniform([], minval=0, maxval=BATCH_SIZE, dtype=tf.int32)
+
+# Get random test image and mask
+test_image = images[random_idx].numpy().astype("float")
+test_mask = masks[random_idx].numpy().astype("float")
+
+pred_image = tf.expand_dims(test_image, axis=0)
+pred_image = keras.applications.vgg19.preprocess_input(pred_image)
+
+# Perform inference on FCN-32S
+pred_mask_32s = fcn32s_model.predict(pred_image, verbose=0).astype("float")
+pred_mask_32s = np.argmax(pred_mask_32s, axis=-1)
+pred_mask_32s = pred_mask_32s[0, ...]
+
+# Perform inference on FCN-16S
+pred_mask_16s = fcn16s_model.predict(pred_image, verbose=0).astype("float")
+pred_mask_16s = np.argmax(pred_mask_16s, axis=-1)
+pred_mask_16s = pred_mask_16s[0, ...]
+
+# Perform inference on FCN-8S
+pred_mask_8s = fcn8s_model.predict(pred_image, verbose=0).astype("float")
+pred_mask_8s = np.argmax(pred_mask_8s, axis=-1)
+pred_mask_8s = pred_mask_8s[0, ...]
+
+# Plot all results
+fig, ax = plt.subplots(nrows=2, ncols=3, figsize=(15, 8))
+
+fig.delaxes(ax[0, 2])
+
+ax[0, 0].set_title("Image")
+ax[0, 0].imshow(test_image / 255.0)
+
+ax[0, 1].set_title("Image with ground truth overlay")
+ax[0, 1].imshow(test_image / 255.0)
+ax[0, 1].imshow(
+    test_mask,
+    cmap="inferno",
+    alpha=0.6,
+)
+
+ax[1, 0].set_title("Image with FCN-32S mask overlay")
+ax[1, 0].imshow(test_image / 255.0)
+ax[1, 0].imshow(pred_mask_32s, cmap="inferno", alpha=0.6)
+
+ax[1, 1].set_title("Image with FCN-16S mask overlay")
+ax[1, 1].imshow(test_image / 255.0)
+ax[1, 1].imshow(pred_mask_16s, cmap="inferno", alpha=0.6)
+
+ax[1, 2].set_title("Image with FCN-8S mask overlay")
+ax[1, 2].imshow(test_image / 255.0)
+ax[1, 2].imshow(pred_mask_8s, cmap="inferno", alpha=0.6)
+
+plt.show()
+
+"""
+## Conclusion
+
+The Fully-Convolutional Network is an exceptionally simple network that has yielded
+strong results in Image Segmentation tasks across different benchmarks.
+With the advent of better mechanisms like [Attention](https://arxiv.org/abs/1706.03762) as used in
+[SegFormer](https://arxiv.org/abs/2105.15203) and
+[DeTR](https://arxiv.org/abs/2005.12872), this model serves as a quick way to iterate and
+find baselines for this task on unknown data.
+"""
+
+"""
+## Acknowledgements
+
+I thank [Aritra Roy Gosthipaty](https://twitter.com/ariG23498), [Ayush
```

Comment:
Make sure that all links are on a single line, unbroken, otherwise they won't render

Reply:
Right, of course, apologies, have corrected that!